src/opers.c

```diff
@@ -1821,2769 +1786,615 @@ static inline Obj CacheOper (
-    return res;
-}
-
-
-/****************************************************************************
-**
-**  DoOperation2Args( <oper>, <a1>, <a2> )
-*/
-Obj DoOperation2Args (
-    Obj                 oper,
-    Obj                 arg1,
-    Obj                 arg2 )
-{
-    Obj                 res;
-    Obj                 type1;
-    Obj                 id1;
-    Obj                 type2;
-    Obj                 id2;
-    Obj *               cache;
-    Obj                 method;
-    Int                 i;
-    Obj                 prec;
-
-    /* get the types of the arguments                                      */
-    type1 = TYPE_OBJ_FEO( arg1 );
-    id1 = ID_TYPE( type1 );
-    type2 = TYPE_OBJ_FEO( arg2 );
-    id2 = ID_TYPE( type2 );
-
-    /* try to find an applicable method in the cache                       */
-    prec = INTOBJ_INT(-1);
-
-    do {
-      /* The next line depends on the implementation of INTOBJS */
-      prec = (Obj)(((Int)prec) +4);
-      method = 0;
-
-      /* recalculate cache each pass, in case of GC     */
-      cache = 1+ADDR_OBJ( CacheOper( oper, 2 ) );
-
-      /* Up to CACHE_SIZE methods might be in the cache */
-      if (prec < INTOBJ_INT(CACHE_SIZE))
-        {
-          for (i = 0;  i < 4*CACHE_SIZE; i+= 4) {
-            if (  cache[i+1] == prec && cache[i+2] == id1
-                  && cache[i+3] == id2 ) {
-              method = cache[i];
-#ifdef COUNT_OPERS
-              OperationHit++;
-#endif
-              break;
-            }
-          }
-        }
-      
-      /* otherwise try to find one in the list of methods                    */
-      if (!method)
-        {
-          if (prec == INTOBJ_INT(0))
-            method = CALL_3ARGS( Method2Args, oper, type1, type2 );
-          else
-            method = CALL_4ARGS( NextMethod2Args, oper, prec, type1, type2 );
-          
-          /* If there was no method found, then pass the information needed for
-             the error reporting. This function rarely returns */
-          if (method == Fail)
-          {
-            Obj args[2];
-            args[0] = arg1;
-            args[1] = arg2;
-            while (method == Fail)
-              method = CallHandleMethodNotFound( oper, 2, (Obj *) args, 0, 0, prec);
-          }
-          
-          /* update the cache */
-          if (method && prec < INTOBJ_INT(CACHE_SIZE))
-            {
-              Bag cacheBag = GET_METHOD_CACHE( oper, 2 );
-              cache = 1+ADDR_OBJ( cacheBag );
-              cache[4*STATE(CacheIndex)] = method;
-              cache[4*STATE(CacheIndex)+1] = prec;
-              cache[4*STATE(CacheIndex)+2] = id1;
-              cache[4*STATE(CacheIndex)+3] = id2;
-              STATE(CacheIndex) = (STATE(CacheIndex) + 1) % CACHE_SIZE;
-              CHANGED_BAG( cacheBag );
-            }
-#ifdef COUNT_OPERS
-          OperationMiss++;
-#endif
+    return method;
+}
+
+static inline void
```

Comment:
It would be great if there was a comment before this function explaining what it is supposed to do, at least very briefly.